### PR TITLE
Sensitive data processing for extended (event) data

### PR DESCRIPTION
**Pull request by djtfmartin**
See: https://github.com/AtlasOfLivingAustralia/extended-data-model/issues/60

These changes remove sensitive records from the elastic index and Spark SQL based downloads.
The following changes where made to achieve this:

* Changes to verbatim-to-interpreted pipeline to apply inheritance at this stage (as opposed to index time) so that inheritance can be used to determine if record is sensitive
* Changes to Elastic pipeline to remove sensitive records
* Generation of a interpreted + verbatim Occurrence output with sensitive rules pre-applied for use in Spark SQL exports

Also includes:

* fixes for NPE for Inherited fields.
* changes to accomodate updated ALA taxonomy in testes
* changes to use hadoop-compress for faster exports

Other issues tackled in this PR:

* #756 
* #858

### CODE REVIEW

File: livingatlas/pipelines/src/main/scala/au/org/ala/pipelines/spark/CmdArgs.scala [hunk 1]
```diff
@@ -0,0 +1,116 @@
+package au.org.ala.pipelines.spark
```
Comment (charvolant) [inline livingatlas/pipelines/src/main/scala/au/org/ala/pipelines/spark/CmdArgs.scala:1]: Was this intended to be in Scala?

File: livingatlas/pipelines/src/main/scala/au/org/ala/pipelines/spark/PredicateExportDwCAPipeline.scala [hunk 2]
```diff
@@ -4,12 +4,11 @@ import au.org.ala.kvs.{ALAPipelinesConfig, ALAPipelinesConfigFactory}
```
Comment (charvolant) [inline livingatlas/pipelines/src/main/scala/au/org/ala/pipelines/spark/PredicateExportDwCAPipeline.scala:4]: Intended to be Scala?

**Review by charvolant — COMMENTED** [review]
A couple of questions about what seems to be Scala in a java-only project.
Happy to be told it's all cool but it looks like something that has snuck in.

File: livingatlas/pipelines/src/main/scala/au/org/ala/pipelines/spark/CmdArgs.scala (hunk 1 above)
Comment (djtfmartin) [reply livingatlas/pipelines/src/main/scala/au/org/ala/pipelines/spark/CmdArgs.scala:1]: Yes

File: livingatlas/pipelines/src/main/scala/au/org/ala/pipelines/spark/PredicateExportDwCAPipeline.scala (hunk 2 above)
Comment (djtfmartin) [reply livingatlas/pipelines/src/main/scala/au/org/ala/pipelines/spark/PredicateExportDwCAPipeline.scala:4]: Yes

File: livingatlas/pipelines/src/test/java/au/org/ala/pipelines/spark/PredicateExportDwCATestIT.java [hunk 3]
```diff
@@ -58,117 +59,127 @@ public void testFullExport() throws Exception {
   private String getDwcaFilePath(String jobId) {
     return "/tmp/la-pipelines-test/event-download/" + jobId + "/dr18391.zip";
   }
-
```
[the remark this answers is not in the document]
Comment (djtfmartin) [reply livingatlas/pipelines/src/test/java/au/org/ala/pipelines/spark/PredicateExportDwCATestIT.java]: good catch - reinstated

File: livingatlas/configs/la-pipelines-local.yaml [hunk 4]
```diff
@@ -1 +1,3 @@
 # which filesystem to use: local or hdfs
+elastic:
+  esSchemaPath: /Users/mar759/dev/pipelines/livingatlas/pipelines/src/test/resources/complete-event-pipeline/es-event-core-schema.json
```
[the remark this answers is not in the document]
Comment (djtfmartin) [reply livingatlas/configs/la-pipelines-local.yaml]: good catch - removed!

**Pull request merged**